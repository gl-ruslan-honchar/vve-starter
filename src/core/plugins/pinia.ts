import type { CustomPlugin } from '~/types'
import { createPinia } from 'pinia'

export const install: CustomPlugin = ({ isClient, initialState, app }) => {
  const pinia = createPinia()
  app.use(pinia)
  if (isClient) pinia.state.value = initialState.pinia || {}
  else initialState.pinia = pinia.state.value
}
