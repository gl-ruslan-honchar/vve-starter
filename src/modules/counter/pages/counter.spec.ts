import { describe, expect, test, beforeEach } from "vitest";
import { mount } from "@vue/test-utils";
import Counter from "./counter.vue";

import { setActivePinia, createPinia } from "pinia";
import { createApp } from "vue";

// you don't need to create one app per test
const app = createApp({});

describe("Counter component", async () => {
  beforeEach(() => {
    const pinia = createPinia()
    app.use(pinia);
    setActivePinia(pinia);
  });

  test("should mount the component", async () => {
    expect(Counter).toBeTruthy();

    const wrapper = mount(Counter);

    expect(wrapper.get('h1').text()).toContain("Simple Counter Test");
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("should update counter text on button click", async () => {
    const wrapper = mount(Counter);

    expect(wrapper.get("span.count").text()).toContain(0);
    
    await wrapper.get("button").trigger("click");

    expect(wrapper.get("span.count").text()).toContain(1);
  });
});
