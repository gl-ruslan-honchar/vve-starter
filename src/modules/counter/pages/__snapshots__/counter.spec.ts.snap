// Vitest Snapshot v1

exports[`Counter component > should mount the component 1`] = `
"<div>
  <h1>Simple Counter Test</h1>
  <div>
    <!-- class=\\"px-6 py-2 bg-gray-500 font-medium text-sm hover:bg-gray-600 text-gray-100 rounded\\" --><button class=\\"btn\\"> Click to update the count: </button>
    <div class=\\"py-4\\"><span class=\\"count\\">0</span></div>
  </div>
</div>"
`;
